src/Contract/Transaction.purs

```diff
@@ -208,6 +291,54 @@ balanceTx
   -> Contract r (Either BalanceTxError.BalanceTxError UnattachedTransaction)
 balanceTx = wrapContract <<< BalanceTx.balanceTx
 
+balanceTxs
+  :: forall
+       (t :: Type -> Type)
+       (r :: Row Type)
+   . (Traversable t)
+  => t UnattachedUnbalancedTx
+  -> Contract r (t UnattachedTransaction)
+balanceTxs uts = do
+  -- First, lock all the already fixated inputs on all the unbalanced transactions.
```

Comment:
```suggestion
  -- First, lock all the already fixed inputs on all the unbalanced transactions.
```

Reply:
If the transaction is not used in the end, the user would either have to call `unlockTransactionInputs` or restart `runContract`  with a new `ContractConfig`. I could perhaps add a (e.g.) `withTransaction` helper function in the style of `bracket`, so that its clear that the inputs get unlocked afterwards.


How would you do it? 

For the idea behind the locking during balancing, see this thread:

https://mlabs-corp.slack.com/archives/C02QXGRGAPK/p1653320841294759

Reply:
I think a `bracket`-style helper would be a great addition to this

Reply:
I added a bunch of them. I'm not sure though... is it too much @ngua @klntsky ?

```purescript
withBalancedTxs
  :: forall (a :: Type)
       (t :: Type -> Type)
       (r :: Row Type)
   . Traversable t
  => t UnattachedUnbalancedTx
  -> (t UnattachedTransaction -> Contract r a)
  -> Contract r a

withBalancedTx
  :: forall (a :: Type)
       (r :: Row Type)
   . UnattachedUnbalancedTx
  -> (UnattachedTransaction -> Contract r a)
  -> Contract r a

withBalancedAndSignedTxs
  :: forall (r :: Row Type) (a :: Type)
   . Array UnattachedUnbalancedTx
  -> (Array BalancedSignedTransaction -> Contract r a)
  -> Contract r a

withBalancedAndSignedTx
  :: forall (a :: Type)
       (r :: Row Type)
   . UnattachedUnbalancedTx
  -> (BalancedSignedTransaction -> Contract r a)
  -> Contract r a
```